src/pages/Home.tsx

```diff
@@ -0,0 +1,234 @@
+
+  const onDrop = useCallback((acceptedFiles: File[]) => {
+    setFile(acceptedFiles[0]);
+    const fr = new FileReader();
+    fr.readAsText(acceptedFiles[0]);
+    fr.onload = () => console.log(fr.result);
+  }, []);
+
+  const { getRootProps, getInputProps, isDragActive } = useDropzone({ onDrop });
+
+  const { submit } = useSubmitTransaction({
+    functionName: 'initializeSarcophagus',
+    contractInterface: EmbalmerFacet__factory.abi,
+  });
+
+  function secondEncryptAndGenerateShards() {
+    const wallet = ethers.Wallet.createRandom();
+    const signingKey = new utils.SigningKey(wallet.privateKey);
+    const publicKey = signingKey.publicKey;
+    console.log(publicKey);
+    setRandomPublicKey(publicKey);
+
+    const secret = Buffer.from(wallet.privateKey);
+    const shards = split(secret, { shares: 3, threshold: minimumNumberShards });
+
+    console.log('shards', shards);
+
+    setArchaeologist([
+      {
+        archAddress: unnamedAccounts[0],
+        bounty: BigNumber.from('100'),
+        diggingFee: BigNumber.from('5'),
+        storageFee: BigNumber.from('10'),
+        hashedShard: utils.id(shards[0].toString('hex')),
+      },
+      {
+        archAddress: unnamedAccounts[1],
+        bounty: BigNumber.from('120'),
+        diggingFee: BigNumber.from('6'),
+        storageFee: BigNumber.from('13'),
+        hashedShard: utils.id(shards[1].toString('hex')),
+      },
+      {
+        archAddress: unnamedAccounts[2],
+        bounty: BigNumber.from('130'),
+        diggingFee: BigNumber.from('4'),
+        storageFee: BigNumber.from('9'),
+        hashedShard: utils.id(shards[2].toString('hex')),
+      },
+    ]);
+  }
+
+  function initializeSarcophagus() {
+    const args = [
+      sarcophagusName,
+      archaeologists,
+      arweaveArchaeologist,
+      recipientAddress,
+      resurrectionTime,
+      maxResurrectionInterval,
+      canBeTransferred,
+      minimumNumberShards,
+      sarcoId,
+    ];
+
+    console.log('initializeSarcophagus args', args);
+    submit({
+      args: args,
+      toastText: 'Initialize Sarcophagus',
+    });
+  }
+  return (
+    <FormControl>
+      <Tabs>
+        <TabList>
+          <Tab>Page 1</Tab>
+          <Tab>Page 2</Tab>
+        </TabList>
+        <TabPanels>
+          <TabPanel>
+            <VStack align="left">
+              <FormLabel>
+                Sarcophagus Name
+                <Input
+                  id="sarcophagusName"
+                  value={sarcophagusName}
+                  onChange={e => {
+                    setSarcophagusName(e.target.value);
+                  }}
+                />
+              </FormLabel>
+              <FormLabel>
+                Recipient
+                <Input
+                  id="recipientAddress"
+                  value={recipientAddress || ''}
+                  onChange={e => {
+                    setRecipientAddress(e.target.value);
+                  }}
+                />
```

Comment:
@mobiusTripper-crypto -- What is the purpose of this recipient address? This could be derived directly from the the public key -- but do we need this?

Reply:
Absolutely. I'll put up a fix for this